subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -219,9 +226,21 @@ static void aws_fota_cb_handler(enum aws_fota_evt_id evt)
 		aws_iot_notify_event(&aws_iot_evt);
 #endif
 		break;
+	case AWS_FOTA_EVT_ERASE_PENDING:
+		LOG_DBG("AWS_FOTA_EVT_ERASE_PENDING");
```

Comment:
At this point you should do `lte_lc_offline` if not the AWS_FOTA process could be stuck in deleting the modem firmware since the modem won't have time to process the delete modem firmware bank request.

Reply:
We don't have this state handling in the AWS Fota sample but you may want to reconnect the MQTT client on the `AWS_FOTA_EVT_ERASE_DONE` event if the connection is dropped since you are shutting down the LTE link.

Reply:
Extended the generic ``cloud`` API and ``aws_iot`` header file in order to propagate these events to the app. Then the app can make the ``link_controller`` related decisions.

Reply:
Okei that's fine by me :)